### PR TITLE
Add top categories block

### LINKED ISSUE #696: Component: Dashboard Leaderboards Block
This description is lifted from the dashboard overview post at: p6riRB-3qG-p2

The leaderboards block allows users to display “top” lists from various reports. It’s a light version of the Tabular data component. The block features a settings menu that allows users to show or hide individual graphs, change the number of rows (3, 5, or 10) and remove the block.

screen shot 2018-10-25 at 8 02 44 pm

Available leaderboards

**Top products
Top categories**
Top coupons

Clicking each product title in the top selling products card will link to the product detail report. As detailed reports do not exist for categories and coupons, clicking the title of a category, or a coupon code will link to the respective reports, filtered to only show the selected category or coupon.

### LINKED ISSUE #1182: Dashboard: Top Selling Products Updates
The Top Selling Products block on the dashboard needs a few more features to be fully complete:

image

- [x] Use the selected Date options from the Date Picker to populate the data.
- [x] Move `select` usage to `wc-api` data layer [#](https://github.com/woocommerce/wc-admin/blob/master/client/dashboard/top-selling-products/index.js#L126)
- [x] Fix missing Product names in the first column
- [ ] Link Orders Quantity to the orders report, for the same date range, with the proper `product_includes` advanced filter set

### LINKED ISSUE #1192: Dashboard: Create Top Categories Block
This task involves creating a new "Top Categories" Block for the dashboard leaderboard section:

image

The table should show the categories with the most items sold:

Category|Items Sold|G. Revenue
---------|-----------|-----------
Category Name linked Product Listing|# of Products sold from category in period|Revenue of Products sold

@LevinMedia should we link the category name to the product listing for that category `/wp-admin/edit.php?s&post_status=all&post_type=product&action=-1&product_cat=CATEGORY-NAME`

### PR DESCRIPTION
**Pull request by emdashcodes**
Fixes #1192. Fixes #1182. Fixes #696.

This PR adds a top selling categories block to the dashboard. It also updates the top selling products block to correctly link to the products report.

### Accessibility

- [x] I've tested using only a keyboard (no mouse)
- [x] I've tested using a screen reader

### Screenshots

screen shot 2019-01-10 at 3 14 08 pm

screen shot 2019-01-10 at 3 37 03 pm

### Detailed test instructions:

* View the dashboard and verify you see your top selling categories.
* Click through to the reports on both the top products and top categories blocks. The dashboard date should be persisted on the report, and the report should be filtered by the single object.

--

Note: Right now the top category block is linking to `compare-categories`. This should actually be a single category filter like the top products block -- but this filter doesn't exist yet so I've added a TODO and opened an issue. #1277.

### CODE REVIEW

**Review by LevinMedia — COMMENTED** [review]
@justinshreve Looking good!

Couple comments: 

lets remove the word "selling" from the labels, that should shorten it up: 

Top Products - Items Sold
Top Categories - Items Sold

We need update the names in the popout to match:

screen shot 2019-01-10 at 3 58 22 pm

These tables shouldn't include the ellipsis menus in their headers that lets you hide and show columns.

We should also remove the "Orders" column from the best selling products leaderboard. 

 Per the new issue #1277 - actually the best way to address that in the categories report would be using the filter function in the header of the tabular data (Still needs to be built, now tracked in #1102) But I think what you have with categories works fine until that lands.

File: client/dashboard/leaderboards/index.js [hunk 1]
```diff
@@ -129,11 +130,13 @@ class Leaderboards extends Component {
 				<div className="woocommerce-dashboard__dashboard-leaderboards">
 					<SectionHeader title={ __( 'Leaderboards', 'wc-admin' ) } menu={ this.renderMenu() } />
 					<div className="woocommerce-dashboard__columns">
-						<div>
-							{ ! hiddenLeaderboardKeys.includes( 'top-products' ) && (
-								<TopSellingProducts query={ this.props.query } totalRows={ rowsPerTable } />
-							) }
-						</div>
+						{ ! hiddenLeaderboardKeys.includes( 'top-products' ) && (
+							<TopSellingProducts query={ this.props.query } totalRows={ rowsPerTable } />
```
Comment (timmyc) [inline client/dashboard/leaderboards/index.js]: Minor but now that we are using `this.props.query` multiple times, maybe we should just assign it to a const.

File: client/dashboard/leaderboards/top-selling-products.js [hunk 2]
```diff
@@ -60,12 +61,22 @@ export class TopSellingProducts extends Component {
 	}
 
 	getRowsContent( data ) {
+		const { query } = this.props;
+		const persistedQuery = getPersistedQuery( query );
 		return map( data, row => {
 			const { product_id, items_sold, net_revenue, orders_count, extended_info } = row;
 			const name = get( extended_info, [ 'name' ] );
+
+			const productUrl = getNewPath( persistedQuery, 'analytics/products', {
```
Comment (timmyc) [inline client/dashboard/leaderboards/top-selling-products.js]: Thanks for fixing this up!

**Comment by timmyc** [comment]
> These tables shouldn't include the ellipsis menus in their headers that lets you hide and show columns.

Indeed I should have created an issue around that when i was working on the leaderboard block. Since that will require touching the base table components, I'm thinking we should handle that in a follow-up PR.

**Review by timmyc — APPROVED** [review]
Minor nit in the code, but this is testing out great for me. Thanks for picking up that last task in the top products too 🙌 pre-approving, but am happy to do another pass if you would like.

**Comment by timmyc** [comment]
Opened up #1280 to take care of the ellipsis menu.

**Pull request merged**